zh/before-start_en.md

```diff
@@ -0,0 +1,62 @@
+# Before the beginning
+## The target audience for this document
+### The confused software developers
+* Developers who want to write code that can be easily read but don't know how to do that  
+* Developers who want to understand other people's code  
+### Software development beginner
+This is a document for software development beginners, who can be defined as having learned no more than two languages and being able to complete most of exercises in the course in terms of the language skills they have learned. From the general situation of college education, they could basicly complete the final course assignments arranged by teachers. We may not be able to say that they have done very well. It is also difficult to improve further.  
+
+For students who have graduated from college(plus or minus one or two years), it is difficult for them to get access to truly complex projects, and they will be incharge of the detailed parts even if they are involved in complex projects. They are eager to improve their software development ability quickly, but by accident, what they lack most is the opportunity to improve their ability.
+
+Therefore, this paper hopes to introduce a more reasonable method to help classmates to improve the ability of software development in a more scientific and reasonable way.
+### First-time users of open source software
+In the field of software development, it is almost impossible not to touch open source projects at all. How to improve the development efficiency with the help of open source projects in the daily development workand reduce the duplication of effort and how to benefit from open source rather than suffer from it is also an interesting topic. In a nutshell, it is risky to use the open source software without reading the source code(there is no different from using normal closed source software or closed source libraries). But if we can read and understand the source code, we can avoid this risk,and the better we understand it, the better we can use the software. Open source software offers this possibility to us.
+
+This article also hopes to help those who are new to open source and want to make good use of it by sharing experience of the experienced person.
+### Hobbyists wandering outside the open source community
+Do you want to be part of the open source community and give back some help to the open source community after enjoying the benefits it offerd? What is going on in the open source community? There are a lot of people who have a natural affinity for open source and hope to join in, integrate into the open source community faster, make more contributions to open source project, or even start their own open source project and pull up a team to do something amazing. There should be many friends that have such idea. We hope the introduction of this article can have certain help.

```

Comment:
Do you want to be part of the open source community and give back some help to the open source community after enjoying the benefits it offered? What is going on in the open source community? There are a lot of people who have a natural affinity for open source and hope to join in. So maybe we can integrate into open source community faster, make more contributions to open source project, or even start their own open source project and pull up a team to do something amazing. There should be many friends that have such idea. We hope the introduction of this article can have certain help.